src/utils/sqs_client.ts

```diff
@@ -0,0 +1,47 @@
+import { SQS } from 'aws-sdk';
+
+/**
+ * SqsClient wraps SQS, making it far easier to unit test SQS and ignore SQS details
+ */
+export class SqsClient {
+    private readonly _sqs: SQS;
+    private readonly _queueUrl: string;
+    constructor(params: { sqs: SQS; queueUrl: string }) {
+        this._sqs = params.sqs;
+        this._queueUrl = params.queueUrl;
+    }
+
+    public async receiveMessageAsync(): Promise<SQS.Message | null> {
+        const response = await this._sqs
+            .receiveMessage({
+                MaxNumberOfMessages: 1,
+                WaitTimeSeconds: 20, // long polling
+                QueueUrl: this._queueUrl,
+            })
+            .promise();
+
+        if (response === undefined || response.Messages?.length !== 1) {
```

Comment:
So it looks like the `client` is the Lower level component, and the `consumer` is the higher level wrapper that is responsible for the lifecycle of the queue. Is that right?

Reply:
Correct - the `client` exists because it was extremely frustrating to write unit tests mocking out `AWS` itself, so I didn't bother and created a wrapper around it. This allows for the `consumer` logic to be well unit tested